src/db_state.rs

```diff
@@ -371,7 +377,24 @@ impl DbState {
         assert!(Arc::ptr_eq(&popped, &imm_memtable));
         state.core.l0.push_front(sst_handle);
         state.core.last_compacted_wal_sst_id = imm_memtable.last_wal_id();
+
+        // in the case of wal disabled, the clock tick is only updated when
+        // we flush to l0 - so we need to update the clock tick and check that
+        // it is valid here (in the case of wal enabled, the manifest tick is
+        // updated when a memtable is flushed)
+        if !wal_enabled {
```

Comment:
This smells funny to me. At first I thought it should go in `mem_table_flush.rs`. But I see it would be unsafe to allow users to move imm memtables without updating the tick. Can we follow a similar pattern to the way we do it here: "in the case of wal enabled, the manifest tick is updated when a memtable is flushed"?

Reply:
Just to be clear, currently the invariant is that the manifest clock tick reflects the latest clock up until `next_sst_wal_id` (which is why it necessitates this difference between wal_enabled/disabled). You are suggesting changing that invariant so that the clock tick in the manifest reflects up until `last_compacted_wal_sst_id`?

I'm OK with that, but I'm worried it may open up some edge cases because now the manifest no longer reflects the complete state of everything it's seen up until then.

Reply:
> You are suggesting changing that invariant so that the clock tick in the manifest reflects up until last_compacted_wal_sst_id?

That does seem more intuitive to me. `last_compacted_clock_tick` or something. I am not at all sure that this is a good idea, though. I am definitely an amateur in this part of the code. WDYT?

Reply:
It certainly simplifies the code, I don't think there's any immediate issues with it so long as we're consistent about the way we use the manifest. There may be some interesting implications for the snapshotting/db cloning depending on how we handle WAL in those situations. So long as we copy over all WALs up until `next_sst_wal_id` I think we should be fine.